jadx-gui/src/main/java/jadx/gui/search/SearchManager.java

```diff
@@ -0,0 +1,39 @@
+package jadx.gui.search;
+
+import java.util.regex.Pattern;
+
+public class SearchManager {
+	/**
+	 * Generate the good pattern according to the different boolean
+	 *
+	 * @param exp           the searched expression
+	 * @param caseSensitive boolean
+	 * @param wholeWord     boolean
+	 * @param useRegexp     boolean
+	 * @return the pattern
+	 */
+	public static Pattern generatePattern(String exp, boolean caseSensitive, boolean wholeWord, boolean useRegexp) {
+		String word = exp;
+		if (word != null && !word.isEmpty()) {
+			if (!useRegexp) {
+				word = word.replace("\\E", "\\E\\\\E\\Q");
+				word = "\\Q" + word + "\\E";
+				if (wholeWord && exp.matches("\\b.*\\b")) {
```

Comment:
Why is the second condition needed here? `exp.matches("\\b.*\\b")`? Isn't just the `wholeWord` flag enough?